tidy3d/components/geometry.py

```diff
@@ -3105,8 +3106,108 @@ def _surface_area(self, bounds: Bound) -> float:
         return area
 
 
+class CustomSurfaceGeometry(Geometry, ABC):
+    """Custom surface geometry given by a triangle mesh, as in the STL file format.
+
+    Example
+    -------
+    >>> from stl import mesh
+    >>> # import STL
+    >>> stl_mesh = mesh.Mesh.from_file("import.stl")
+    >>> geometry = CustomSurfaceGeometry.from_stl(stl_mesh.vectors)
+    >>> # export STL
+    >>> vectors = geometry.to_stl()
+    >>> stl_mesh = mesh.Mesh(np.zeros(len(vectors), dtype=mesh.Mesh.dtype))
+    >>> stl_mesh.vectors = vectors
+    >>> stl_mesh.save("export.stl")
+    """
+
+    triangle_data: TriangleDataArray = pydantic.Field(
+        ...,
+        title="Surface triangle vertices",
+        description="Data array consisting of surface triangle vertices.",
+    )
+
+    @classmethod
+    def from_stl(cls, triangle_vertices: ArrayLike[3]) -> CustomSurfaceGeometry:
```

Comment:
we talked earlier about passing the `stl.mesh.Mesh` object directly, similar to how `PolySlab.from_gds()` accepts a `gdstk.Cell`. Does it make more sense to accept a raw array (which could be passed directly to `triangle_data`) or a `Mesh` object? I think with the `Mesh` object being passed, we could add additional options to filter the data (eg by material / layer), which could come in handy. Also it hides any additional work converting their `Mesh` to raw vertices. Thoughts?

Reply:
For exporting, I can't export an stl.mesh.Mesh because I don't want to add numpy-stl as a dependency. So the export still has these extra steps. Any suggestions here?